app/app_test.go

```diff
@@ -5420,17 +5420,22 @@ func (s *S) TestUpdateMetadataWhenEmpty(c *check.C) {
 	err := CreateApp(context.TODO(), &app, s.user)
 	c.Assert(err, check.IsNil)
 
-	updateData := App{Metadata: appTypes.Metadata{
+	expectedMetadata := appTypes.Metadata{
 		Annotations: []appTypes.MetadataItem{{Name: "a", Value: "b"}},
 		Labels:      []appTypes.MetadataItem{{Name: "c", Value: "d"}},
-	}}
-	err = app.Update(context.TODO(), UpdateAppArgs{UpdateData: updateData, Writer: new(bytes.Buffer)})
+	}
+
+	updateData := App{Metadata: expectedMetadata}
+	err = app.Update(context.TODO(), UpdateAppArgs{UpdateData: updateData, Writer: new(bytes.Buffer), ShouldRestart: true})
 	c.Assert(err, check.IsNil)
 
 	dbApp, err := GetByName(context.TODO(), app.Name)
 	c.Assert(err, check.IsNil)
-	c.Assert(dbApp.Metadata.Annotations, check.DeepEquals, []appTypes.MetadataItem{{Name: "a", Value: "b"}})
-	c.Assert(dbApp.Metadata.Labels, check.DeepEquals, []appTypes.MetadataItem{{Name: "c", Value: "d"}})
+
+	expectedMetadataJSON, _ := json.Marshal(expectedMetadata)
```

Comment:
for those cases you could add a `c.Assert(err, check.IsNil)` as a way to not ignore Marshal output